exporters/otlp/otlpmetric/internal/transform/metricdata.go

```diff
@@ -174,17 +174,24 @@ func HistogramDataPoints(dPts []metricdata.HistogramDataPoint) []*mpb.HistogramD
 	out := make([]*mpb.HistogramDataPoint, 0, len(dPts))
 	for _, dPt := range dPts {
 		sum := dPt.Sum
-		out = append(out, &mpb.HistogramDataPoint{
+		hdp := &mpb.HistogramDataPoint{
 			Attributes:        AttrIter(dPt.Attributes.Iter()),
 			StartTimeUnixNano: uint64(dPt.StartTime.UnixNano()),
 			TimeUnixNano:      uint64(dPt.Time.UnixNano()),
 			Count:             dPt.Count,
 			Sum:               &sum,
 			BucketCounts:      dPt.BucketCounts,
 			ExplicitBounds:    dPt.Bounds,
-			Min:               dPt.Min,
-			Max:               dPt.Max,
-		})
+		}
+		if dPt.Min.Valid {
+			min := dPt.Min.Value
+			hdp.Min = &min
```

Comment:
Same concern here: can we refer to `&dPts[index].Min.Value?`. This value `dPt` is a loop-local variable and I'm concerned that all data points will point to the same min value. (Apologies if I'm reading incorrectly.)

Reply:
The reference is copied here `min := dPt.Min.Value` to prevent the values from all being the same.